Lib/selectors.py

```diff
@@ -580,16 +581,31 @@ def close(self):
             super().close()
 
 
+def _can_use(method):
+    """Check if we can use the selector depending upon the
+    operating system. """
+    # Implementation based upon https://github.com/sethmlarson/selectors2/blob/master/selectors2.py
+    try:
+        if method == 'poll':
+            poll_obj = select.poll()
+            poll_obj.poll(0)
+        else:
+            getattr(select, method)().close()
+        return True
+    except (OSError, AttributeError):
+        return False
+
+
 # Choose the best implementation, roughly:
 #    epoll|kqueue|devpoll > poll > select.
 # select() also can't accept a FD > FD_SETSIZE (usually around 1024)
-if 'KqueueSelector' in globals():
+if _can_use('kqueue'):
     DefaultSelector = KqueueSelector
-elif 'EpollSelector' in globals():
-    DefaultSelector = EpollSelector
-elif 'DevpollSelector' in globals():
+elif _can_use('devpoll'):
     DefaultSelector = DevpollSelector
-elif 'PollSelector' in globals():
+elif _can_use('epoll'):
+    DefaultSelector = EpollSelector
+elif _can_use('poll'):
     DefaultSelector = PollSelector
-else:
+elif hasattr(select, 'select'):
```

Comment:
AFAIK all supported platforms have a ``select`` call.

Reply:
Yes, I added that check for safe guarding. @tiran Thanks for review.

Reply:
I agree that the `hasattr()` check is a bit redundant and not needed for safeguarding since we don't currently support any platforms without `select` support. There are other stdlib modules (including asyncio) which make the assumption that support for `select` is present, so the `elif hasattr(select, 'select')` just adds extra overhead (compared to `else`) without really safeguarding anything. AFAIK, if a platform does not support select, it would not work with Python stdlib anyways.

It would be one thing if that check was already present and we were discussing removing it (potentially breaking systems that presently work), but I'm opposed to arbitrarily adding the safeguard. So, I would request that it be changed back to the existing `else:`.